mozilla-release/mobile/android/tests/cliqz/java/com/cliqz/mobile/test/CliqzRunner.java

```diff
@@ -0,0 +1,83 @@
+package com.cliqz.mobile.test;
+
+import android.util.Log;
+
+import org.junit.Ignore;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.internal.runners.model.EachTestNotifier;
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runner.notification.StoppedByUserException;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.Statement;
+
+public class CliqzRunner extends BlockJUnit4ClassRunner {
+
+    private static final int RETRY_COUNT = 3;
+
+    public CliqzRunner(Class<?> clazz) throws InitializationError {
+        super(clazz);
+    }
+
+    @Override
+    public void run(final RunNotifier notifier) {
+        EachTestNotifier testNotifier = new EachTestNotifier(notifier, getDescription());
+        Statement statement = classBlock(notifier);
+        try {
+            statement.evaluate();
+        } catch (AssumptionViolatedException ave) {
+            testNotifier.fireTestIgnored();
+        } catch (StoppedByUserException sbue) {
+            throw sbue;
+        } catch (Throwable t) {
+            Log.d("AUTOBOTS", "Retry class: " + getDescription().getDisplayName());
```

Comment:
Would `Log.e()` or `Log.w()` tell the severity better here?

https://developer.android.com/studio/debug/am-logcat#format

Reply:
Makes sense to use `Log.w()`. Done.